src/lib/sql/sql_pipeline_statement.cpp

```diff
@@ -16,8 +15,20 @@
 #include "sql/sql_translator.hpp"
 #include "utils/assert.hpp"
 
+#if HYRISE_JIT_SUPPORT
+#include "logical_query_plan/jit_aware_lqp_translator.hpp"
+#else
+#include "logical_query_plan/lqp_translator.hpp"
+#endif
+
 namespace opossum {
 
+#if HYRISE_JIT_SUPPORT
+using CurrentLQPTranslator = JitAwareLQPTranslator;
+#else
+using CurrentLQPTranslator = LQPTranslator;
```

Comment:
This should be a runtime setting if JIT support is enabled. Maybe discuss this with @mrzzzrm

Reply:
I never liked that compile-time mechanism either, but it kind of worked for me so far. Where would that setting come from and how would it get propagated to the pipeline statement? @mrzzzrm Any ideas?

Reply:
You could make `_use_jit` a parameter of the SQLPipeline/SQLPipelineBuilder and then pick the correct Translator at runtime, while Assert()ing, that `_use_jit` is only set to true when `HYRISE_JIT_SUPPORT`.

Reply:
I added an `lqp_translator` parameter to SQLPipeline/SQLPipelineBuilder analog to the optimizer and also added a check of `HYRISE_JIT_SUPPORT ` to the constructor of the jit-aware translator.